### PR TITLE
P2P GetEvents

### PR DESCRIPTION
**Pull request by kirugan**
Issue #1072

### CODE REVIEW

File: p2p/starknet/adapters.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package starknet
```
Comment (omerfirmak) [inline p2p/starknet/adapters.go:1]: can we move these to `adapter` folder? similar to what I have done [here](https://github.com/NethermindEth/juno/pull/1086/commits/f0b3dee8aeb3ecbe369e6e274082a805fe54a460)

File: p2p/starknet/handlers.go [hunk 2]
```diff
@@ -143,3 +160,15 @@ func (h *Handler) HandleGetTransactions(req *spec.GetTransactions) (*spec.Transa
 		Transactions: make([]*spec.Transaction, magic),
 	}, nil
 }
+
+func (h *Handler) blockByID(id *spec.BlockID) (*core.Block, error) {
+	switch {
+	case id == nil:
+		return nil, errors.New("block id is nil")
+	case id.Hash != nil:
+		hash := new(felt.Felt).SetBytes(id.Hash.Elements)
```
Comment (omerfirmak) [inline p2p/starknet/handlers.go:169]: Let's do this in an adaptation layer to be consistent? ie something like `p2p2core.AdaptHash`

File: p2p/starknet/adapters.go (hunk 1 above)
Comment (kirugan) [reply p2p/starknet/adapters.go]: Yep, moved.

File: p2p/starknet/handlers.go (hunk 2 above)
Comment (kirugan) [reply p2p/starknet/handlers.go]: Done

File: p2p/starknet/handlers_test.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+			height := uint64(777)
+			someErr := errors.New("some error")
+			mockReader.EXPECT().BlockByNumber(height).Return(nil, someErr)
+
+			_, err := h.HandleGetEvents(&spec.GetEvents{
+				Id: &spec.BlockID{
+					Height: height,
+				},
+			})
+			assert.True(t, errors.Is(err, someErr))
```
Comment (joshklop) [inline p2p/starknet/handlers_test.go:58]:
```suggestion
			assert.ErrorIs(t, err, someErr)
```
I might be mis-remembering the name, but I'm pretty sure there is a helper for this.

File: p2p/starknet/handlers_test.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		sl[i] = randFelt()
+	}
+
+	return sl
+}
+
+func randFelt() *felt.Felt {
+	f, err := new(felt.Felt).SetRandom()
+	if err != nil {
+		panic(err)
```
Comment (joshklop) [inline p2p/starknet/handlers_test.go:126]: Can we pass in `*testing.T` and `require.NoError` here?

**Review by joshklop — APPROVED** [review]

File: p2p/starknet/handlers_test.go (hunk 3 above)
Comment (kirugan) [reply p2p/starknet/handlers_test.go]: Thanks, added.

File: p2p/starknet/handlers_test.go (hunk 4 above)
Comment (kirugan) [reply p2p/starknet/handlers_test.go]: Yep, added

**Pull request merged**